core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1145,4 +1146,14 @@ default MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession
     {
         return Optional.empty();
     }
+
+    /**
+     * Canonicalizes the provided SQL identifier according to connector-specific rules
+     * for the purpose of providing the name in metadata APIs
+     */
+    default String canonicalize(ConnectorSession session, String identifier, boolean delimited)
+    {
+        // TODO: Move to model which is in accordance with SQL specification
```

Comment:
I don't get this TODO, can you please elaborate?

Reply:
The default implementation in `ConnectorMetadata` should be as per SQL specification... Like non-delimited identifier should be converted to Upper case instead of lower case.